### DISCUSSION TITLE
v0.23 Perhaps form struct tags should default to "-" and not ""

### DISCUSSION THREAD

**Discussion by dschissler** [pocketbase/pocketbase#5717] [Q&A]
I noticed that http request form body parsing behavior changed from echo to the 0.23 refactor.   I'm thinking that it's more secure to have to explicitly declare the field tag for form parsing instead of it defaulting to something (lower case I guess).

One possibility 1) is that you have to explicitly default the struct tag with explicit name, another is that 2) you have to at least put `form` on the field to have it try to figure out the name of it.

Current 0.23 requirement:

```golang
type SiteCreate struct {
	Name            string       `form:"name"`
	Owner           *core.Record `form:"-"`
	Tier            *core.Record `form:"-"`
}
```

 Alternative that is compatible with echo)

```golang
type SiteCreate struct {
	Name            string       `form:"name"`
	Owner           *core.Record
	Tier            *core.Record
}
```

I can change this to have setters for all non-form fields by making them non-exported lower case.  It's just different though and more code.

At the very least there should be documentation to let people know that some of their forms might be breaking (depending upon how they made them).

**Comment by ganigeorgiev** [comment]
If your struct have public fields then even with echo you can scan value in them. Note that how and which struct tag will be used also depends on the request Content-Type (e.g. if it is json then it runs `json.Unmarshal` on the struct and doesn't care for the `form` tag; it should be the same for echo).

I don't think there is need to change the current behavior. If you have struct fields that you don't want to unmarshal data into then it would be better to either unexport them or create a separate DTO with only the fields that are safe to be scanned.

_But I agree, I'll consider adding a note in the routing docs for this._

**Reply by dschissler** [reply]
One of my forms changed behavior, which is enough to raise attention to it.  I'll unexport a few fields and add setters.

**Reply by ganigeorgiev** [reply]
Again, PocketBase supports both `application/json` and `multipart/form-data` requests (_technically also xml but it behaves similar to the json unmarshaling_). Even without the explcit form tag, if users are submitting json payload it will still try to populate your struct.

I agree that there are some minor difference between the new router and echo but there is no difference regarding the security aspect between the two. Even echo recommends to use separate DTO to avoid accidentally scanning data into public fields - https://echo.labstack.com/docs/binding#security (_I'll consider adding a similar note in the new routing docs_)

**Reply by dschissler** [reply]
Alright, I'll my forms to have another scanning struct.
